webui/vite.config.js

```diff
@@ -1,6 +1,9 @@
-import reactRefresh from '@vitejs/plugin-react-refresh';
+import react from '@vitejs/plugin-react';
 import eslintPlugin from 'vite-plugin-eslint';
 import replace from '@rollup/plugin-replace';
+import dns from "dns";
+
+dns.setDefaultResultOrder('verbatim');
```

Comment:
Not sure we need this one - the first request is the minimum nodejs version it requires and the fact that I don't have a problem dns resolved 'localhost' on local development.
Did you had an issue with the default dns resolution?

Reply:
@nopcoder I didn't have an issue either until the Vite v3 upgrade. Without this Vite mounts on `127.0.0.1` and does not resolve `localhost` correctly (at least on my machine - node 16.16.0.)  Come to think of it, while checking this, I also fixed the default port (back to 3000), so I'll double-check the need for modifying the DNS resolution locally again to be sure it's needed.

I see that the GitHub agent is running node 16.17.1. I'll `nvm` that locally and see if it's a node version issue or a GitHub agent memory issue.

Reply:
Also, probably a good time to add a `.nvmrc` file to declare the preferred node version

Reply:
From what I read from the docs the issue is in case your dns localhost resolve don't not point to 127.0.0.1 the request may go to a different server - think it is a rare case and I am fan of using defaults.

Reply:
nvm is a specific tool - prefer to specify it in package.json using:
```json
 "engines" : { 
    "node" : ">=16.0.0"
  }
```

Reply:
Good point. I'll do that.

Reply:
@nopcoder Fixed + tests are now passing